sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java

```diff
@@ -80,6 +81,9 @@ public Duration getDuration() {
      */
     @Beta(value = Beta.SinceVersion.V4_9_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     public Set<URI> getRegionsContacted() {
+        if (this.feedResponseDiagnostics != null) {
+            return null;
```

Comment:
Wouldn't you need to get the superset of the FeedRangeDiagnostics.getClientSideRequestStatistics().getRegionsContacted() here instead?

Reply:
Aggregating regions contacted from all list of clientSideRequestStatistics needs to be done.  I will do this in subsequent PRs